src/app/hackernews-api.service.ts

```diff
@@ -1,28 +1,46 @@
 import { Injectable } from '@angular/core';
-import { Http } from '@angular/http';
+// import { Http } from '@angular/http';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
 
+declare var fetch;
+
+//wrap fetch in observable so we can keep it chill
+export function lazyFetch(url, options?){
```

Comment:
This is awesome. Thank you so much! 

Only thing I noticed is that with this set up, the `onCompleted`doesn't fire in `stories.component.ts` when `fetchStories` is triggered. See [here](https://github.com/hdjirdeh/angular2-hn/blob/master/src/app/stories/stories.component.ts#L38-L39).

This means that the list numbering of items isn't correct, window doesn't scroll up when navigating as well as the checks on the `prev` and `next` (can click `prev` and `next` continuously with no checks in place). We just need to fix how we wrap  the completed event.